lib/plugin.js

```diff
@@ -94,20 +94,16 @@ ManifestPlugin.prototype.apply = function(compiler) {
 
     var json = JSON.stringify(cache, null, 2);
 
-    compilation.assets[outputName] = {
-      source: function() {
-        return json;
-      },
-      size: function() {
-        return json.length;
-      }
-    };
-
-    if (this.opts.writeToFileEmit) {
-      var outputFolder = compilation.options.output.path;
-      var outputFile = path.join(outputFolder, this.opts.fileName);
+    var outputFolder = compilation.options.output.path;
+    var outputFile = path.join(outputFolder, this.opts.fileName);
 
+    const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem' ||
```

Comment:
I agree it's ugly but that the only way I could find to check if it's really `MemoryFileSystem`, if you have a better solution feel free to share it ;)

I'm note the only one doing it: https://github.com/gajus/write-file-webpack-plugin/blob/51167a8cf33ee0885805abd1fb85c8306f0eda27/src/index.js#L19